specification/securityinsights/resource-manager/Microsoft.SecurityInsights/stable/2023-11-01/SourceControls.json

```diff
@@ -0,0 +1,1005 @@
+        "values": [
+          {
+            "value": "OAuth"
+          },
+          {
+            "value": "PAT"
+          },
+          {
+            "value": "App"
+          }
+        ]
+      }
+    },
+    "Version": {
+      "readOnly": true,
+      "description": "The version of the source control.",
+      "enum": [
+        "V1",
+        "V2"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "modelAsString": true,
+        "name": "Version",
+        "values": [
+          {
+            "value": "V1"
+          },
+          {
+            "value": "V2"
+          }
+        ]
+      }
+    },
+    "DeploymentFetchStatus": {
+      "description": "Status while trying to fetch the deployment information.",
+      "enum": [
+        "Success",
+        "Unauthorized",
+        "NotFound"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "modelAsString": true,
+        "name": "DeploymentFetchStatus",
+        "values": [
+          {
+            "value": "Success"
+          },
+          {
+            "value": "Unauthorized"
+          },
+          {
+            "value": "NotFound"
+          }
+        ]
+      }
+    },
+    "DeploymentState": {
+      "description": "The current state of the deployment.",
+      "enum": [
+        "In_Progress",
+        "Completed",
+        "Queued",
+        "Canceling"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "modelAsString": true,
+        "name": "DeploymentState",
+        "values": [
+          {
+            "value": "In_Progress"
+          },
+          {
+            "value": "Completed"
+          },
+          {
+            "value": "Queued"
+          },
+          {
+            "value": "Canceling"
+          }
+        ]
+      }
+    },
+    "DeploymentResult": {
+      "description": "Status while trying to fetch the deployment information.",
+      "enum": [
+        "Success",
+        "Canceled",
+        "Failed"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "modelAsString": true,
+        "name": "DeploymentResult",
+        "values": [
+          {
+            "value": "Success"
+          },
+          {
+            "value": "Canceled"
+          },
+          {
+            "value": "Failed"
+          }
+        ]
+      }
+    },
+    "PullRequest": {
+      "readOnly": true,
+      "description": "Information regarding pull request for protected branches.",
+      "type": "object",
+      "properties": {
+        "url": {
+          "description": "URL of pull request",
+          "type": "string",
+          "readOnly": true
+        },
+        "state": {
+          "$ref": "#/definitions/PullRequestState",
+          "description": "State of the pull request",
+          "type": "object",
+          "readOnly": true
+        }
+      }
+    },
+    "PullRequestState": {
+      "description": "Status of the pull request.",
+      "enum": [
+        "Open",
```

Comment:
Are there other states like InReview, Draft etc.?

Reply:
No, we only care about those states that are mentioned.